Detector dev nid assets

dniklaus opened the pull request:
This shall resolve #31 (by workaround)
- Assets with fake persistent memory to store LoRa Session Keys (currently for 6 nodes)
- LoraWanAbp gets the Session Keys configured during setup by data from Assets

mareklinka committed:
Add Zurich weather history data

NiaX84 committed:
script for json to csv conversion

NiaX84 committed:
Merge branch 'master' of https://github.com/ERNICommunity/dust-measurement-network

wtvisser committed:
Add files via upload

wtvisser committed:
Add files via upload

Historic weather data for Zurich

wtvisser committed:
Historic weather data for Stuttgart 2018

wtvisser committed:
Delete Stuttgart_weather_historic_2018.xlsx

NiaX84 committed:
data aggregation

NiaX84 committed:
renaming of the aggregated file

NiaX84 committed:
removing old file

Commit:
Add kubernetes deployment

NiaX84 committed:
add interleaved data

NiaX84 committed:
Merge branch 'master' of https://github.com/ERNICommunity/dust-measurement-network

NiaX84 committed:
add date-time transformation

NiaX84 committed:
combining interleaving, binning and date-trensformation into one python script: interleave.py. Storing final training set into data_to_use/ folder

mareklinka committed:
Add Keras model

mareklinka committed:
Categorize more features, add comments

NiaX84 committed:
first refactoring. Adding on script that runs the full data augmentation. Helper scripts in filders

NiaX84 committed:
Merge branch 'master' of https://github.com/ERNICommunity/dust-measurement-network

mareklinka committed:
Improve dust level filtering

mareklinka committed:
Improve aggregation with multisensor support

mareklinka committed:
Fix reset condition, remove BOM from weather CSV

NiaX84 committed:
Merge pull request #35 from ERNICommunity/lima/filtering

Improve filtering and aggregating

mareklinka committed:
Train model on latest aggregation

mareklinka committed:
Fix training data preparation

mareklinka committed:
Add Flask server

mareklinka committed:
Reorganize code and data

mareklinka committed:
Add dockerfile

mareklinka committed:
Refactoring + new training result

NiaX84 committed:
Merge pull request #36 from ERNICommunity/lima/filtering

Prediction server + refactoring

mareklinka committed:
Fix shuffling inconsistency

Retrain on correctly shuffled data

dniklaus committed:
Assets: add LoRa Key access stubs

dniklaus committed:
Assets: add Persistent Data Memory interface

dniklaus committed:
Assets: add fake Persistent Data Memory for Detector

dniklaus committed:
Assets: prepare handling of Persistent Data Memory

dniklaus committed:
Assets: integrate memory access

dniklaus committed:
Assets: add LoRa Keys debug CLI command

dniklaus committed:
Assets: rename m_deviceSerialNr attribute of DetectorFakePersDataMemory

dniklaus committed:
Assets: fix device serial number update

dniklaus committed:
main.cpp: inject DetectorFakePersDataMemory as Assets' persistent data memory

dniklaus committed:
LoraWanAdapter: get rid of compiler warnings

dniklaus committed:
add LoraWan/ILoraWanConfigAdapter.h

dniklaus committed:
Assets: correct captions for values in Get LoRaKeys DbgCmd

dniklaus committed:
LoraWan/LoraWanAbp.cpp: set TODO bookmark

dniklaus committed:
make LoraWanAbp configurable (work in progress)

dniklaus committed:
make LoraWanAbp configurable (work in progress)

dniklaus committed:
apply LoRaWan config during setup

aschoepfer commented on detector/lib/LoraWan/ILoraWanConfigAdapter.h:
```diff
@@ -0,0 +1,44 @@
+
+  /**
+   * Read Network Session Key byte array.
+   * @param Network Session Key OUT: Buffer for Network Session Key byte array
```
nwkSKey

dniklaus replied:
THX!

aschoepfer commented on detector/lib/assets/Assets.cpp:
```diff
@@ -13,17 +13,23 @@
 #include <DbgTracePort.h>
```
Evtl noch Tests ergänzen für diese Assets.

dniklaus replied:
Yes, indeed, unit tests are missing.

aschoepfer reviewed (approved):
Hoi Didi
Saubere Arbeit, habe gerade keine Fehler entdeckt. Müsste mich aber wieder mal gründlich in den Detector einarbeiten um alle Details zu verstehen.
Lieber Gruss Andreas

dniklaus commented on detector/lib/assets/AssetsDbgCmd.cpp:
```diff
@@ -65,14 +66,69 @@ void AssetsDbgCmd_GetDeviceSerial::execute(unsigned int argc, const char** args,
-      Serial.println("Assets - Device Serial Number:");
-      if (0 != m_assets->getDeviceSerialNrAdapter())
+      Serial.println("Assets - LoRa Keys:");
+      if (0 != m_assets->getPersistentDataMemory())
```
```suggestion
      if (0 != m_assets->getDeviceSerialNrAdapter())
```

dniklaus commented on detector/lib/adapters/MyLoRaWanConfigAdapter.h:
```diff
@@ -0,0 +1,61 @@
+
+  /**
+   * Read Application Session Key byte array.
+   * @param Application Session Key OUT: Buffer for Application Session Key byte array
```
```suggestion
    * @param appSKey Application Session Key OUT: Buffer for Application Session Key byte array
```

dniklaus commented on detector/lib/adapters/MyLoRaWanConfigAdapter.h:
```diff
@@ -0,0 +1,61 @@
+
+  /**
+   * Read Network Session Key byte array.
+   * @param Network Session Key OUT: Buffer for Network Session Key byte array
```
```suggestion
    * @param nwkSKey Network Session Key OUT: Buffer for Network Session Key byte array
```

dniklaus commented on detector/lib/LoraWan/ILoraWanConfigAdapter.h:
```diff
@@ -0,0 +1,44 @@
+
+  /**
+   * Read Application Session Key byte array.
+   * @param Application Session Key OUT: Buffer for Application Session Key byte array
```
```suggestion
    * @param appSKey Application Session Key OUT: Buffer for Application Session Key byte array
```

dniklaus commented on detector/lib/LoraWan/ILoraWanConfigAdapter.h:
```diff
@@ -0,0 +1,44 @@
+
+  /**
+   * Read Network Session Key byte array.
+   * @param Network Session Key OUT: Buffer for Network Session Key byte array
```
```suggestion
    * @param nwkSKey Network Session Key OUT: Buffer for Network Session Key byte array
```

dniklaus reviewed (commented):
ready to merge

dniklaus committed:
Apply suggestions from code review

dniklaus commented on detector/lib/assets/AssetsDbgCmd.cpp:
```diff
@@ -65,14 +66,69 @@ void AssetsDbgCmd_GetDeviceSerial::execute(unsigned int argc, const char** args,
     if (0 != m_assets)
     {
-      Serial.println("Assets - Device Serial Number:");
+      Serial.println("Assets - LoRa Keys:");
```
```suggestion
       Serial.println("Assets - Device Serial Numbe:");
```

dniklaus committed:
fix detector/hterm-com10.cfg: DTR on

dniklaus committed:
Merge branch 'detector-dev-nid-assets' of github.com:ERNICommunity/dust-measurement-network into detector-dev-nid-assets

dniklaus committed:
FIX AssetsDbgCmd multiple bad issues :-O

dniklaus committed:
LoraWan/LoraWanAdapter: add DbgTrace_Port and accessor to it

dniklaus committed:
LoraWan/LoraWanAbp: try to get back the formerly running hardcoded LMIC configuration

dniklaus committed:
CHG: rename LoraWanAdapter to LoRaWanDriver
NEW: LoRaWanDriver confuration takes LoRa security keys and Device Address from Assets
NEW: add single channel mode to configuration (can be enabled by 'dbg lora sch' debug cli command)

dniklaus committed:
deactivate keys in LoraWan/configuration.h

dniklaus committed:
add missing dbg cli command classes

Commit:
remove unused code

dniklaus committed:
Merge branch 'master' into detector-dev-nid-assets
